pkg/karmadactl/cmdinit/karmada/check.go

```diff
@@ -14,19 +15,39 @@ import (
 
 // WaitAPIServiceReady wait the api service condition true
 func WaitAPIServiceReady(c *aggregator.Clientset, name string, timeout time.Duration) error {
-	if err := wait.PollImmediate(time.Second, timeout, func() (done bool, err error) {
-		apiService, e := c.ApiregistrationV1().APIServices().Get(context.TODO(), name, metav1.GetOptions{})
-		if e != nil {
-			return false, nil
+	apiService, err := c.ApiregistrationV1().APIServices().Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
```

Comment:
What if the APIService not exist at that time?

Reply:
I think when this situation occurs, the normal error can be thrown, WaitAPIServiceReady premise is that there is already APIService

Reply:
Please @lonelyCZ check if the behavior change is acceptable.

Reply:
Before executing `WaitAPIServiceReady`, we firstly `CreateOrUpdateAPIService` in general.

https://github.com/karmada-io/karmada/blob/360412399b494d903576e0dba7afe78ea3dbf37a/pkg/karmadactl/cmdinit/karmada/deploy.go#L302-L308